src/Entities/Boat.cpp

```diff
@@ -169,3 +169,105 @@ void cBoat::UpdatePaddles(bool a_RightPaddleUsed, bool a_LeftPaddleUsed)
 
 	m_World->BroadcastEntityMetadata(*this);
 }
+
+
+
+
+
+cBoat::eMaterial cBoat::ItemToMaterial(const cItem & a_Item)
+{
+	switch (a_Item.m_ItemType)
+	{
+		case E_ITEM_BOAT:          return bmOak;
+		case E_ITEM_SPRUCE_BOAT:   return bmSpruce;
+		case E_ITEM_BIRCH_BOAT:    return bmBirch;
+		case E_ITEM_JUNGLE_BOAT:   return bmJungle;
+		case E_ITEM_ACACIA_BOAT:   return bmAcacia;
+		case E_ITEM_DARK_OAK_BOAT: return bmDarkOak;
+		default:
+		{
+			LOGWARNING("%s: Item type not handled %d.", __FUNCTION__, a_Item.m_ItemType);
+			return cBoat::bmOak;
+		}
+	}
+}
+
+
+
+
+
+AString cBoat::MaterialToString(eMaterial a_Material)
+{
+	switch (a_Material)
+	{
+		case bmOak:     return "oak";
+		case bmSpruce:  return "spruce";
+		case bmBirch:   return "birch";
+		case bmJungle:  return "jungle";
+		case bmAcacia:  return "acacia";
+		case bmDarkOak: return "dark_oak";
+	}
+	#ifndef __clang__
+		return "oak";
```

Comment:
The common practice is to add an `ASSERT` here - we want to catch this if it ever happens in MSVC.